pkg/util/credential.go

```diff
@@ -57,20 +62,20 @@ func ObtainCredentialsFromMemberCluster(clusterKubeClient kubeclient.Interface,
 	var impersonatorSecret *corev1.Secret
 	var clusterSecret *corev1.Secret
 	var err error
-	// It's necessary to set the label of namespace to make sure that the namespace is created by Karmada.
-	labels := map[string]string{
-		ManagedByKarmadaLabel: ManagedByKarmadaLabelValue,
-	}
 	// ensure namespace where the karmada control plane credential be stored exists in cluster.
-	if _, err = EnsureNamespaceExistWithLabels(clusterKubeClient, opts.ClusterNamespace, opts.DryRun, labels); err != nil {
+	if _, err = EnsureNamespaceExistWithLabels(clusterKubeClient, opts.ClusterNamespace, opts.DryRun, managedByKarmadaLabels); err != nil {
 		return nil, nil, err
 	}
 
 	if opts.IsKubeImpersonatorEnabled() {
 		// create a ServiceAccount for impersonation in cluster.
-		impersonationSA := &corev1.ServiceAccount{}
-		impersonationSA.Namespace = opts.ClusterNamespace
-		impersonationSA.Name = names.GenerateServiceAccountName("impersonator")
+		impersonationSA := &corev1.ServiceAccount{
```

Comment:
https://github.com/karmada-io/karmada/blob/adef1e59748e1e1d31cb75fffe406b5dd69a66d7/pkg/util/serviceaccount.go#L104-L114
Does the `secret` here also need to be handled in the same way?

Reply:
Nice finding! I think Yes.

Reply:
/cc @a7i

Reply:
Kindly ping @a7i

Reply:
nice catch, let me fix that and rebase

Reply:
Implemented ✅